R/html_document.R

```diff
@@ -179,6 +180,7 @@ html_document <- function(toc = FALSE,
                           toc_depth = 3,
                           toc_float = FALSE,
                           number_sections = FALSE,
+                          anchor_sections = FALSE,
```

Comment:
What about making it a default ? I find it to be very pratical. Do we think this would be a too impactful change for the community ?

Reply:
All the more if the anchor icon only appear on hover. It is rather discrete

Reply:
I'm fine with make it a default.
The current decision is based on other features, which are mostly `FALSE` by default.

@yihui What do you think?

Reply:
While I'd really love to have anchors by default (they are absolutely helpful), my only concern is for people who have already added the anchors by themselves, and they won't want the extra anchors that we provide.

Thinking about this problem, I feel it might be a better idea to implement this via JavaScript instead of a Lua filter:  https://yihui.org/en/2018/11/md-js-tricks/#how-to-add-anchor-links-to-section-headings With JavaScript, you may be able to detect if anchors already exist (at least make sure **pkgdown** and https://desiree.rbind.io/post/2019/bookdown-anchor-links/ still work), and add them only if they have not already been added. However, we also need to make sure our JavaScript executes _after_ their anchors have been added. This might be a little tricky.